File: 项目文件/frontend/src/types/vote.ts
export interface Vote {
  id: string;
  title: string;
  description: string | null;
  options: string[];
  allow_multiple: boolean;
  status: string;
  deadline: string | null;
  owner_id: string;
  visibility: string;
  restricted_users: string[] | null;
  restricted_tags: string[] | null;
  created_at: string;
  updated_at: string;
}

export interface VoteCreate {
  title: string;
  description?: string;
  options: string[];
  allow_multiple?: boolean;
  deadline?: string;
  visibility?: 'public' | 'private' | 'restricted';
  restricted_users?: string[] | undefined;
  restricted_tags?: string[] | undefined;
}

export interface VoteListResponse {
  items: Vote[];
  total: number;
}

export interface VoteResult {
  option: string;
  count: number;
  percentage: number;
}

File: 项目文件/backend/app/schemas/vote.py
"""投票 Schema"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

from app.core.visibility import Visibility


class VoteCreate(BaseModel):
    title: str
    description: str | None = None
    options: list[str]
    allow_multiple: bool = False
    deadline: str | None = None
    visibility: Visibility = Visibility.PRIVATE
    restricted_users: list[str] | None = None
    restricted_tags: list[str] | None = None


class VoteResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str
    description: str | None
    options: list[str]
    allow_multiple: bool
    status: str
    deadline: datetime | None
    owner_id: uuid.UUID
    visibility: str
    restricted_users: list[str] | None = None
    restricted_tags: list[str] | None = None
    created_at: datetime
    updated_at: datetime


class VoteListResponse(BaseModel):
    items: list[VoteResponse]
    total: int


class VoteSubmit(BaseModel):
    selected_options: list[str]


class VoteResult(BaseModel):
    option: str
    count: int
    percentage: float
